### DISCUSSION TITLE
iPhone SMS to Firefly-iii transaction (Solution)

### DISCUSSION THREAD

**Discussion by mrahmadt** [firefly-iii/firefly-iii#8833] [General]
Hello

I'm working on a solution to automate converting iPhone SMS to transaction (for my personal use), I need it to automate the transaction creation, otherwise I'm toooo lazy to add them manually.

The solution contain the following parts

1- iOS message filter extension (developed and was able to to receive SMS to my server!)
2- SMS Receiver (PHP), receive the SMS from iOS message filter extension and determine if it's bank transaction or not (Developed)
3- Parse SMS and extract transaction information such as amount, source, destination... (in-progress)
4- Send the transaction information to firefly-iii via API to the correct account (pending)

the process is straightforward, but unfortunately it's not something that will work out of the box for everyone!, if you want to use it you will need the following
- Apple Developer account, and be able to build the code and run the iOS message filter extension to your phone (technically it's not possible to package it and push it to Apple Store, the Firefly-iii has to be hard coded in the extension)
- Web server, domain name with SSL certificate 
- PHP programming to create your bank parser for the SMS messages

With all the above prerequisites, are you interested in such tool? should I publish it once I'm finish or just keep it for my personal use?

if I publish it, I need to spend some time to document it and to make the code generic for anyone to use (which take time form me), this is why I'm asking.

**Comment by JC5** [comment]
I believe you should publish, absolutely! I'm a great fan of open source work and yours could inspire others to build something similar.

**Comment by mrahmadt** [comment]
iOS part of the solution has been tested and published 🚀 https://github.com/mrahmadt/SMSReceiver-iOS

Backend (SMS Parsing) under testing (will be published to https://github.com/mrahmadt/smartMoney)

**Reply by JC5** [reply]
Cool! 🎉

**Comment by mrahmadt** [comment]
Backend published to Github https://github.com/mrahmadt/smartMoney

Just need to code few small features add how-to docs 

- Automatic SMS Parsing and Transaction Creation
- Simple Budget Web Application
- Alert Notifications
- Detection of Abnormal Transactions
- Automatic Subscription Detection
- Subscription Amount Increase Alerts

**Reply by JC5** [reply]
Nice, I've expanded the entry in the documentation :+1:
